src/google/Gantt.js

```diff
@@ -0,0 +1,209 @@
+            this._data_google.addColumn('string', this.columns()[6]);
+
+            var parseDate = d3.time.format(this.datePattern()).parse;
+            
+            data = this.data().map(function(d) {
+                var label = d[0];
+                var start = parseDate(d[1][0]);
+                var end = parseDate(d[1][1]);
+                var duration = d[2] || (end - start) || null;
+                var id = d[3] || d[0];
+                var resource = d[4];
+                var completeness = d[5];
+                var dependency = d[6];
+
+                if (duration) {
+                    switch (this.durationUnit()) {
+                        case "day":
+                            duration *= 24 * 60 * 60 * 1000;
+                            break;
+                        case "hour":
+                            duration *= 60 * 60 * 1000;
+                            break;
+                        case "minute":
+                            duration *= 60 * 1000;
+                            break;
+                        case "second":
+                            duration *= 1000;
+                            break;
+                        default:
+                            break;
+                    }
+                }
+                
+                this._data_google.addRows([ [id, label, resource, start, end, duration, completeness, dependency] ]);
+            }, this);
+        } else {
+              data = [
+                ["", { role: "annotation" }],
+                ["", ""]
+            ];
+        }
+        return this._data_google;
+    };
+
+    Gantt.prototype.click = function (row, column, selected) {
+        console.log("Click:  " + JSON.stringify(row) + ", " + column + ", " + selected);
+    };
+
+    Gantt.prototype.enter = function (domNode, element) {
+        HTMLWidget.prototype.enter.apply(this, arguments);
+        element.style("overflow", "hidden");
+        this._chart = new google.visualization[this._chartType](domNode);
+
+        var context = this;
+        google.visualization.events.addListener(this._chart, "select", function () {
+            var selectedItem = context._chart.getSelection()[0];
+            if (selectedItem) { // without this double clicking a selection throws an error
+                context._selection = {
+                    data: context.rowToObj(context.data()[selectedItem.row] || {}),
+                    column: context.columns()[selectedItem.column] || null
+                };
+            }
+            context.click(context._selection.data, context._selection.column, Object.keys(context._selection).length !== 0);
+        });
+    };
+
+    Gantt.prototype.update = function (domNode, element) {
+        HTMLWidget.prototype.update.apply(this, arguments);
+        this._chart.draw(this.formatData(), this.getChartOptions());
+    };
+
+    function materialHack(callback, depth) {
```

Comment:
Duplicate code - can you common this and the one in the Material bar chart up to a "material" base class.